drivers/mpu9250/MPU9250.hpp

```diff
@@ -54,21 +181,70 @@ namespace DriverFramework
 
 #define MPU_WHOAMI_9250			0x71
 
+#pragma pack(push, 1)
+struct fifo_packet
+{
+	int16_t accel_x;
+	int16_t accel_y;
+	int16_t accel_z;
+	int16_t temp;
+	int16_t gyro_x;
+	int16_t gyro_y;
+	int16_t gyro_z;
+	//uint8_t		ext_data[24];
+};
+struct fifo_packet_with_mag
+{
+	int16_t accel_x;
+	int16_t accel_y;
+	int16_t accel_z;
+	int16_t temp;
+	int16_t gyro_x;
+	int16_t gyro_y;
+	int16_t gyro_z;
+	char mag_st1; // 14 mag ST1 (1B)
+	int16_t mag_x; // 15-16 (2B)
+	int16_t mag_y; // 17-18 (2B)
+	int16_t mag_z; // 19-20 (2B)
+	char mag_st2; // 21 mag ST2 (1B)
+};
+// This data structure is a copy of the segment of the above fifo_packet_with_mag data
+// struture that contains mag data.
+struct mag_data
+{
+	char mag_st1; // mag ST1 (1B)
+	int16_t mag_x; // uT (2B)
+	int16_t mag_y; // uT (2B)
+	int16_t mag_z; // uT (2B)
+	char mag_st2; // mag ST2 (1B)
+};
+#pragma pack(pop)
 
-class MPU9250 : public ImuSensor
+class MPU9250: public ImuSensor
 {
 public:
-	MPU9250(const char *device_path) :
-		ImuSensor(device_path, MPU9250_MEASURE_INTERVAL_US),
-		_last_temp_c(0.0f),
-		_temp_initialized(false)
+	MPU9250(const char *device_path, bool mag_enabled = false) :
+			ImuSensor(device_path, MPU9250_MEASURE_INTERVAL_US, mag_enabled), // true = mag is enabled
+			_last_temp_c(0.0f), _temp_initialized(false),
+			_mag_enabled(mag_enabled)
 	{
 		m_id.dev_id_s.devtype = DRV_DF_DEVTYPE_MPU9250;
 		// TODO: does the WHOAMI make sense as an address?
 		m_id.dev_id_s.address = MPU_WHOAMI_9250;
 	}
 
 	// @return 0 on success, -errno on failure
+	int writeReg(int reg, uint8_t val)
```

Comment:
Yes.  They expose what would otherwise be a protected member function in SPIDevObj as a public member function callable by the mag object.